universal-application-tool-0.0.1/app/views/admin/questions/QuestionsListView.java

```diff
@@ -0,0 +1,95 @@
+package views.admin.questions;
+
+import static j2html.TagCreator.a;
+import static j2html.TagCreator.body;
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.each;
+import static j2html.TagCreator.table;
+import static j2html.TagCreator.tbody;
+import static j2html.TagCreator.th;
+import static j2html.TagCreator.thead;
+import static j2html.TagCreator.tr;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import j2html.tags.Tag;
+import play.twirl.api.Content;
+import services.question.QuestionDefinition;
+import views.BaseHtmlLayout;
+import views.BaseHtmlView;
+
+public final class QuestionsListView extends BaseHtmlView {
+  private final BaseHtmlLayout layout;
+  private final ImmutableList<QuestionTableCell> tableCells =
+      ImmutableList.of(
+          QuestionTableCell.ID,
+          QuestionTableCell.VERSION,
+          QuestionTableCell.PATH,
+          QuestionTableCell.NAME,
+          QuestionTableCell.DESCRIPTION,
+          QuestionTableCell.QUESTION_TEXT,
+          QuestionTableCell.QUESTION_HELP_TEXT,
+          QuestionTableCell.QUESTION_TYPE,
+          QuestionTableCell.ACTIONS);
+
+  @Inject
+  public QuestionsListView(BaseHtmlLayout layout) {
+    this.layout = layout;
+  }
+
+  /** Renders a page with either a table or a list view of all questions. */
+  public Content render(ImmutableList<Tag> questionContent) {
```

Comment:
question (blocking): should this be a private method? Should any controller be allowed to call this render method and pass in whatever tags it wants?

Reply:
Done